lib/streamlit/elements/legacy_data_frame.py

```diff
@@ -322,14 +312,21 @@ def has_custom_display_value(cell):
                 raise RuntimeError('Failed to parse cell selector "%s"' % cell_id)
 
             # Only store display values that differ from the cell's default
-            if has_custom_display_value(cell):
+            if _has_custom_display_value(cell):
                 row = int(match.group(1))
                 col = int(match.group(2))
                 display_values[(row, col)] = str(cell["display_value"])
 
     return display_values
 
 
+def _has_custom_display_value(cell: Dict[Any, Any]) -> bool:
+    """True if a cell's display_value differs from its stringified value."""
+    value = cell["value"]
+    display_value = cell["display_value"]
+    return str(value) != str(display_value)
+
+
```

Comment:
This is the actual functionality change. We're just checking to see if a dataframe cell's `display_value` is different from its stringified `value`.

The previous version of this function was local to the `_get_custom_display_values` function, above. It did a more "thorough" comparison of display_value vs value - but I don't think any of that is actually necessary. Ultimately, what we care about is whether we should send over a custom string for a cell's display value, and the only time that's important is when that custom string will be different from the default value string.